Main/OOP_HW16/coffee_shop.cpp

```diff
@@ -0,0 +1,71 @@
+#include "coffee_shop.h"
+#include <iostream>
+#include <utility>
+
+CoffeeShop::CoffeeShop(std::string shopName) : name(std::move(shopName)) {
+    for (TABLE_DSIZE i = 0; i < TABLES_COUNT; ++i) {
+        m_Tables[i] = false;
+        m_Orders[i] = nullptr;
+    }
+}
+
+int CoffeeShop::takePlace() {
+    for (TABLE_DSIZE i = 0; i < TABLES_COUNT; ++i) {
+        if (!m_Tables[i]) {
+            m_Tables[i] = true;
+            return i;
+        }
+    }
+    return -1;
+}
+
+void CoffeeShop::order(int tableNumber) {
+    if (tableNumber < 0 || tableNumber >= TABLES_COUNT || !m_Tables[tableNumber]) {
+        std::cout << "Invalid table number.\n";
+        return;
+    }
```

Comment:
I would split it in two.
The first one is invalid table
And the second one is table already taken.
Theoretically, these two cases will need two different handings